### PR TITLE
feat(mesh): add collection page

### PR DESCRIPTION
**Pull request by caesarWHLee**
# Features & Changes

- Add collection page - [figma](https://www.figma.com/design/wqODc69zqgbu7TZFtQAmnN/2024_READr-Mesh_Web?node-id=13082-324314&t=oR0elajMDcl7QiNj-4) / [skeleton figma](https://www.figma.com/design/wqODc69zqgbu7TZFtQAmnN/2024_READr-Mesh_Web?node-id=9942-136692&t=WsHZ8p7sb0Vg7aQx-4)
- Support picking stories and collections
- Support commenting on stories and collections
- Support bookmarking stories and collections (pub/sub not support bookmark collection yet)
- BottomActionBar used on story and collection page (both article and default layout)
- Profile page image blur on local server

## For Reviewing

Currently there are no way to add/edit collection, to review this PR reviewer can check the existing collection in CMS and open the /collection/${collectionId} to check the page

### CODE REVIEW

File: packages/mesh/hooks/use-picker.ts [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+  const [isLoading, setIsLoading] = useState(false)
+  const memberId = user.memberId
+  const { addToast } = useToast()
+
+  const addPick = useCallback(
+    async (targetId: string, pickObjective: PickObjective) => {
+      if (!memberId) return
+      setIsLoading(true)
+
+      const reverseMutation = addPickToUser(
```
Comment (tyc67) [inline packages/mesh/hooks/use-picker.ts:27]: addPick reverse 應該是 `removePickFromUser` ?

File: packages/mesh/hooks/use-picker.ts [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    },
+    [memberId, user, setUser, addToast]
+  )
+
+  // TODO: simplify the mutation and toast message
+  const removePick = useCallback(
+    async (targetId: string, pickObjective: PickObjective) => {
+      if (!memberId) return
+      setIsLoading(true)
+      const reverseMutation = removePickFromUser(
```
Comment (tyc67) [inline packages/mesh/hooks/use-picker.ts:53]: removePick reverse 應該是 `addPickToUser` ?

File: packages/mesh/hooks/use-picker.ts [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    },
+    [memberId, user, setUser, addToast]
+  )
+
+  const addPickAndComment = useCallback(
+    async (targetId: string, pickObjective: PickObjective, comment: string) => {
+      if (!memberId || !comment) return
+      setIsLoading(true)
+
+      const reverseMutation = addPickToUser(
```
Comment (tyc67) [inline packages/mesh/hooks/use-picker.ts:91]: addPickAndComment reverse 應該是 `removePickFromUser` ?

File: packages/mesh/components/share-sheet.tsx [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+] as const
+
+export default function ShareSheet({
+  url,
+  onClose,
+}: {
+  url: string
+  onClose: () => void
+}) {
+  const getShareUrl = (urlTemplate: string) => {
```
Comment (tyc67) [inline packages/mesh/components/share-sheet.tsx]: 可以整合至 `get-url.ts`

File: packages/mesh/app/collection/[id]/_components/collection-pick-count.tsx [hunk 5]
```diff
@@ -0,0 +1,24 @@
+export default function CollectionPickCount({
```
Comment (tyc67) [inline packages/mesh/app/collection/[id]/_components/collection-pick-count.tsx:2]: 可以考慮和 `collection-comment-count` 整併

**Review by LinHeMa — APPROVED** [review]
LGTM for 共用元件、函式 與 profile, comments part

File: packages/mesh/app/collection/[id]/layout.tsx [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+export default async function CollectionLayout({
+  children,
+  params,
+}: {
+  children: React.ReactNode
+  params: { id: string }
+}) {
+  const collectionId = params.id
+
+  const collectionData = await getCollection({
```
Comment (tyc67) [inline packages/mesh/app/collection/[id]/layout.tsx:18]: 提問: 為何在 `layout` 和 `page` 都做 `getCollection()` data fetching

File: packages/mesh/hooks/use-picker.ts (hunk 1 above)
Comment (caesarWHLee) [reply packages/mesh/hooks/use-picker.ts:27]: 改動在 [d1c5d06](https://github.com/readr-media/Sachiel/pull/889/commits/d1c5d06f11e565d220c99be147fcfe57857bebc0)

File: packages/mesh/hooks/use-picker.ts (hunk 2 above)
Comment (caesarWHLee) [reply packages/mesh/hooks/use-picker.ts:53]: 改動在 [d1c5d06](https://github.com/readr-media/Sachiel/pull/889/commits/d1c5d06f11e565d220c99be147fcfe57857bebc0)

File: packages/mesh/hooks/use-picker.ts (hunk 3 above)
Comment (caesarWHLee) [reply packages/mesh/hooks/use-picker.ts:91]: 改動在 [d1c5d06](https://github.com/readr-media/Sachiel/pull/889/commits/d1c5d06f11e565d220c99be147fcfe57857bebc0)

File: packages/mesh/components/share-sheet.tsx (hunk 4 above)
Comment (caesarWHLee) [reply packages/mesh/components/share-sheet.tsx]: 改動在 [d1c5d06](https://github.com/readr-media/Sachiel/pull/889/commits/d1c5d06f11e565d220c99be147fcfe57857bebc0)

File: packages/mesh/app/collection/[id]/_components/collection-pick-count.tsx (hunk 5 above)
Comment (caesarWHLee) [reply packages/mesh/app/collection/[id]/_components/collection-pick-count.tsx:2]: 先加 TODO: 之後再補 [1a5723a](https://github.com/readr-media/Sachiel/pull/889/commits/1a5723a756aabbd0b3daafc59f7a636cabf76c39)

File: packages/mesh/app/collection/[id]/layout.tsx (hunk 6 above)
Comment (caesarWHLee) [reply packages/mesh/app/collection/[id]/layout.tsx:18]: 因為兩者都會用到且資料無法直接傳遞，使用 next.js server component 機制讓第二次重複 call api 的時候可以直接拿取 cache 的結果

**Review by tyc67 — APPROVED** [review]

**Pull request merged**